TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/activities/RunningActivity.java

```diff
@@ -747,7 +753,35 @@ protected void init(NodeMainExecutor nodeMainExecutor) {
         tangoConfigurationParameters.put(getString(R.string.pref_localization_map_uuid_key), "string");
         mParameterNode = new ParameterNode(this, tangoConfigurationParameters);
         nodeConfiguration.setNodeName(mParameterNode.getDefaultNodeName());
-        nodeMainExecutor.execute(mParameterNode, nodeConfiguration);
+        nodeMainExecutor.execute(mParameterNode, nodeConfiguration, new ArrayList<NodeListener>(){{
+            add(new DefaultNodeListener() {
+                @Override
+                public void onStart(final ConnectedNode connectedNode) {
+                    class SetParameterAndroidApiLevelTask implements Callable<Boolean> {
+                        @Override
+                        public Boolean call() throws Exception{
+                            while (connectedNode.getParameterTree() == null) {
+                                try {
+                                    Thread.sleep(200);
+                                } catch (InterruptedException e) {
+                                    e.printStackTrace();
+                                    return false;
+                                }
+                            }
+                            mParameterNode.setIntParam("android_api_level", Build.VERSION.SDK_INT);
```

Comment:
Can you move the code to set a parameter to the `onStart` method of mParameterNode? This seems a more simple solution, but maybe I am missing something.